addon/components/discover-page/component.js

```diff
@@ -0,0 +1,808 @@
+     * @property {String} start
+     */
+    start: '',
+    /**
+     * Subject query parameter.  If "subject" is one of your query params, it must be passed to the component so it can be reflected in the URL.
+     * @property {String} subject
+     */
+    subject: '',
+    /**
+     * Tags query parameter.  If "tags" is one of your query params, it must be passed to the component so it can be reflected in the URL.
+     * @property {String} tags
+     */
+    tags: '',
+    took: 0,
+    /**
+     * type query parameter.  If "type" is one of your query params, it must be passed to the component so it can be reflected in the URL.
+     * @property {String} type
+     */
+    type: '',
+
+    // ************************************************************
+    // COMPUTED PROPERTIES and OBSERVERS
+    // ************************************************************
+
+    // TODO update this property if a solution is found for the elastic search limitation.
+    // Ticket: SHARE-595
+    clampedPages: Ember.computed('totalPages', 'size', function() {
+        // Total pages of search results, unless total is greater than the max pages allowed.
+        let maxPages = Math.ceil(10000 / this.get('size'));
+        let totalPages = this.get('totalPages');
+        return totalPages < maxPages ? totalPages : maxPages;
+    }),
+    elasticAggregations: Ember.computed(function() {
+        // Ember-SHARE property.
+        return {
+            sources: {
+                terms: {
+                    field: 'sources',
+                    size: MAX_SOURCES
+                }
+            }
+        };
+    }),
+    facetStates: Ember.computed(...filterQueryParams, 'end', 'start', function() {
+        // Ember-SHARE property.  Watches query params in URL and modifies facetStates
+        let facetStates = {};
+        for (let param of filterQueryParams) {
+            facetStates[param] = getSplitParams(this.get(param));
+        }
+        facetStates.date = { start: this.get('start'), end: this.get('end') };
+        return facetStates;
+    }),
+    facetStatesArray: Ember.computed('facetStates', function() { // Modified when query params in URL change.
+        let facets = this.get('facetStates');
+        let facetArray = [];
+        for (let key of Object.keys(facets)) {
+            facetArray.push({ key: key, value: facets[key] });
+        }
+        return facetArray;
+    }),
+    hiddenPages: Ember.computed('clampedPages', 'totalPages', function() {
+        // Ember-SHARE property. Returns pages of hidden search results.
+        const total = this.get('totalPages');
+        const max = this.get('clampedPages');
+        if (total !== max) {
+            return total - max;
+        }
+        return null;
+    }),
+    providerChanged: Ember.on('init', Ember.observer('provider', function() {
+        // For PREPRINTS and REGISTRIES - watches provider query param for changes and modifies activeFilters
```

Comment:
The observer is something brought over from preprints.  I experimented with some other ways of doing this, but right now this is one of the main ways the application is notified that something has changed and search needs to be reloaded.